src/components/shared/Entity/DetailsForm/useEvaluateStorageMapping.ts

```diff
@@ -0,0 +1,108 @@
+import type { DataPlaneOption } from 'src/stores/DetailsForm/types';
+
+import { useCallback } from 'react';
+
+import { useUserInfoSummaryStore } from 'src/context/UserInfoSummary/useUserInfoSummaryStore';
+import { useDetailsFormStore } from 'src/stores/DetailsForm/Store';
+import { useEntitiesStore } from 'src/stores/Entities/Store';
+import { useWorkflowStore } from 'src/stores/Workflow/Store';
+import { getDataPlaneInfo } from 'src/utils/dataPlane-utils';
+
+export const useEvaluateStorageMapping = () => {
+    const options = useDetailsFormStore((state) => state.dataPlaneOptions);
+
+    const storageMappings = useEntitiesStore((state) => state.storageMappings);
+
+    const hasSupportRole = useUserInfoSummaryStore(
+        (state) => state.hasSupportAccess
+    );
+
+    const setStorageMappingPrefix = useWorkflowStore(
+        (state) => state.setStorageMappingPrefix
+    );
+    const currentStorageMappingPrefix = useWorkflowStore(
+        (state) => state.storageMappingPrefix
+    );
+    const previousNameEmpty = useWorkflowStore(
+        (state) => state.catalogName.whole.length === 0
+    );
+
+    const getDataPlaneOption = useCallback(
+        (
+            dataPlaneId: string | undefined,
+            catalogName: string | undefined,
+            dataPlaneNames: string[]
+        ) => {
+            let selectedOption = options.find(
+                (option) => option.id === (dataPlaneId ?? '')
+            );
+
+            if (typeof selectedOption !== 'undefined') {
+                return selectedOption;
+            }
+
+            if (!hasSupportRole && catalogName) {
+                return dataPlaneNames.length > 0
+                    ? options.find(
+                          (option) =>
+                              option.dataPlaneName.whole === dataPlaneNames[0]
+                      )
+                    : undefined;
+            }
+
+            return undefined;
+        },
+        [hasSupportRole, options]
```

Comment:
We should just pass in `hasSupportRole` to the function since it is "private" and the caller already listens to changes on that property.